decidim-core/app/commands/decidim/create_report.rb

```diff
@@ -69,7 +69,11 @@ def hidden_by_admin?
     end
 
     def hideable?
-      hidden_by_admin? || (!@reportable.hidden? && moderation.report_count >= Decidim.max_reports_before_hiding)
+      hidden_by_admin? || hidden_by_spam_engine? || (!@reportable.hidden? && moderation.report_count >= Decidim.max_reports_before_hiding)
+    end
+
+    def hidden_by_spam_engine?
```

Comment:
Let's extract the last statement to it's own method to better document it and to improve legibility. I'm not 100% about the method name though, feel free to improve on it

```suggestion
      hidden_by_admin? || hidden_by_spam_engine? || hidden_by_report count?
    end

    def hidden_by_report count?
      !@reportable.hidden? && moderation.report_count >= Decidim.max_reports_before_hiding
    end

    def hidden_by_spam_engine?
```